### PR TITLE
8297445: PPC64: Represent Registers as values

### PR DESCRIPTION
**Pull request by TheRealMDoerr**
The recent Register implementation uses wild pointer (including null pointer) dereferences which exhibit undefined behavior. We should migrate away from pointer-based representation of Register values as it was done for x86 ([JDK-8292153](https://bugs.openjdk.org/browse/JDK-8292153)). Problems exist when trying to build with GCC 11 ([JDK-8297426](https://bugs.openjdk.org/browse/JDK-8297426)).
Note: Implicit conversion from `intptr_t` to `RegisterOrConstant` is no longer supported. That's why I had to replace some `add` instructions.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8297445](https://bugs.openjdk.org/browse/JDK-8297445): PPC64: Represent Registers as values

### Reviewers
 * [Matthias Baesken](https://openjdk.org/census#mbaesken) (@MBaesken - **Reviewer**) ⚠️ Review applies to [5af4a37b](https://git.openjdk.org/jdk/pull/11297/files/5af4a37b0b2812a7fe904ea49153b41c3de921b2)
 * [Richard Reingruber](https://openjdk.org/census#rrich) (@reinrich - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/11297/head:pull/11297` \
`$ git checkout pull/11297`

Update a local copy of the PR: \
`$ git checkout pull/11297` \
`$ git pull https://git.openjdk.org/jdk pull/11297/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 11297`

View PR using the GUI difftool: \
`$ git pr show -t 11297`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/11297.diff">https://git.openjdk.org/jdk/pull/11297.diff</a>

</details>

### CODE REVIEW

**Review by MBaesken — APPROVED** [review]
Looks good to me and fixes the issue I reported (https://bugs.openjdk.org/browse/JDK-8297426).
You might want to check the SAP copyright header lines in some of the files.

**Review by reinrich — APPROVED** [review]
Clean as a whistle :)
Thanks, Richard.

File: src/hotspot/cpu/ppc/register_ppc.hpp [hunk 1]
```diff
@@ -75,614 +74,414 @@ typedef VMRegImpl* VMReg;
...
+  int _encoding;
  public:
   enum {
     number_of_registers = 32
   };
 
+  constexpr Register(int encoding = -1) : _encoding(encoding) {}
+  bool operator==(const Register rhs) const { return _encoding == rhs._encoding; }
+  bool operator!=(const Register rhs) const { return _encoding != rhs._encoding; }
+  const Register* operator->() const { return this; }
```
[the remark this answers is not in the document]
Comment (TheRealMDoerr) [reply src/hotspot/cpu/ppc/register_ppc.hpp:89]: Correct. I think this should get done at some point of time.

**Comment by TheRealMDoerr** [comment]
Thanks for the reviews!

**Pull request closed without merging**